Mathlib/MeasureTheory/Function/L1Space/Integrable.lean

```diff
@@ -134,27 +184,43 @@ theorem MemLp.integrable_norm_rpow' [IsFiniteMeasure μ] {f : α → β} {p : 
   · simp [h_top, integrable_const]
   exact hf.integrable_norm_rpow h_zero h_top
 
+lemma MemLp.integrable_enorm_pow {f : α → ε} {p : ℕ} (hf : MemLp f p μ) (hp : p ≠ 0) :
+    Integrable (fun x : α ↦ ‖f x‖ₑ ^ p) μ := by
+  simpa using hf.integrable_enorm_rpow (mod_cast hp) (by simp)
+
 lemma MemLp.integrable_norm_pow {f : α → β} {p : ℕ} (hf : MemLp f p μ) (hp : p ≠ 0) :
     Integrable (fun x : α => ‖f x‖ ^ p) μ := by
   simpa using hf.integrable_norm_rpow (mod_cast hp) (by simp)
 
+lemma MemLp.integrable_enorm_pow' [IsFiniteMeasure μ] {f : α → β} {p : ℕ} (hf : MemLp f p μ) :
+    Integrable (fun x : α ↦ ‖f x‖ₑ ^ p) μ := by simpa using hf.integrable_enorm_rpow'
+
 lemma MemLp.integrable_norm_pow' [IsFiniteMeasure μ] {f : α → β} {p : ℕ} (hf : MemLp f p μ) :
     Integrable (fun x : α => ‖f x‖ ^ p) μ := by simpa using hf.integrable_norm_rpow'
 
+lemma integrable_enorm_rpow_iff {f : α → ε} {p : ℝ≥0∞}
+    (hf : AEStronglyMeasurable f μ) (p_zero : p ≠ 0) (p_top : p ≠ ∞) :
+    Integrable (fun x : α => ‖f x‖ₑ ^ p.toReal) μ ↔ MemLp f p μ := by
+  rw [← memLp_enorm_rpow_iff (q := p) hf p_zero p_top, ← memLp_one_iff_integrable,
+    ENNReal.div_self p_zero p_top]
+
 lemma integrable_norm_rpow_iff {f : α → β} {p : ℝ≥0∞}
     (hf : AEStronglyMeasurable f μ) (p_zero : p ≠ 0) (p_top : p ≠ ∞) :
     Integrable (fun x : α => ‖f x‖ ^ p.toReal) μ ↔ MemLp f p μ := by
   rw [← memLp_norm_rpow_iff (q := p) hf p_zero p_top, ← memLp_one_iff_integrable,
     ENNReal.div_self p_zero p_top]
 
-theorem Integrable.mono_measure {f : α → β} (h : Integrable f ν) (hμ : μ ≤ ν) : Integrable f μ :=
+theorem Integrable.mono_measure {f : α → ε} (h : Integrable f ν) (hμ : μ ≤ ν) : Integrable f μ :=
   ⟨h.aestronglyMeasurable.mono_measure hμ, h.hasFiniteIntegral.mono_measure hμ⟩
 
-theorem Integrable.of_measure_le_smul {μ' : Measure α} {c : ℝ≥0∞} (hc : c ≠ ∞) (hμ'_le : μ' ≤ c • μ)
-    {f : α → β} (hf : Integrable f μ) : Integrable f μ' := by
+theorem Integrable.of_measure_le_smul {ε} [TopologicalSpace ε] [ENormedAddMonoid ε]
+    {μ' : Measure α} {c : ℝ≥0∞} (hc : c ≠ ∞) (hμ'_le : μ' ≤ c • μ)
+    {f : α → ε} (hf : Integrable f μ) : Integrable f μ' := by
   rw [← memLp_one_iff_integrable] at hf ⊢
   exact hf.of_measure_le_smul hc hμ'_le
 
+-- TODO: AEStronglyMeasurable.add_measure requires β to be pseudo-metrizable,
```

Comment:
Why can't we add that as an extra hypothesis to the generalized lemma?

Reply:
Sure, we can. But would this actually be more general? Is ENNReal pseudo-metrizable?

Reply:
Actually, let me retract that statement: ENNReal is metrizable, as $$[0,1]\to\mathbb{R}_{\geq 0}\cup\{\infty\}, t \mapsto \frac{1}{1-t}-1$$ is a homeomorphism between them. (It's bijective and strictly increasing, hence a homeomorphism between both spaces with the order topology. But these are the standard topologies we want.)

Reply:
Yes, and Mathlib knows this ([ENNReal.instMetrizableSpace](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Metrizable/Real.html#ENNReal.instMetrizableSpace)). This is just not the "usual" `edist`.
(Is [ENNReal.orderIsoUnitIntervalBirational](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/ENNReal/Inv.html#ENNReal.orderIsoUnitIntervalBirational) precisely the function you wrote?)

Reply:
:bulb: (it's not precisely the function I wrote, but a similar one)